### DISCUSSION TITLE
How to set a layer's material to 0?

### DISCUSSION THREAD

**Discussion by human890209** [sn4k3/UVtools#277] [Scripts]
Hi,
I write a script which copys layers at the same PostionZ. How to set the layer's material consumption to 0ml of those copied layers?
Layer's MaterialMilliliters property can't be SET by script.

**Comment by sn4k3** [comment]
You can't. That property is auto calculated.
Why you want to set to 0?

**Reply by human890209** [reply]
I use erode duplicated layers(Same PositionZ) to make Strong and Detailed results. But the material volume result is wrong. I want to set the duplicated layers' MaterialMillilters to be 0.

**Reply by sn4k3** [reply]
Well that is not possible atm
Material volume is recalculated every time you change the layer image or the layer set. So if you are eroding the image it will recalculate the volume. But you are double exposure a layer with same image?

**Reply by human890209** [reply]
Double exposure with one original image with ExposureTime for the best detail and one eroded image with extra ExposureTime for the best strength. I want to set the extra strengthen layer's material to 0.

**Reply by sn4k3** [reply]
How that work compared if you exposure the layer at once at ExposureTime+extra time?
I can open that property to public but in end the material ml will never be absolute 0, it will always cure some resin even if you use same image on same position

**Reply by sn4k3** [reply]
You can now set that property

**Reply by human890209** [reply]
Cool! Exposuring layers with longer exposureTime without erode causes the elephant foot effect. And any fix for the elephant foot loses details.

**Reply by sn4k3** [reply]
Thats what dim wall is for with Elephant foot calibration on UVtools

https://github.com/sn4k3/UVtools/wiki/Calibration:-Elephant-Foot
https://www.facebook.com/groups/uvtools/posts/526030405173134/

**Reply by human890209** [reply]
By the way, the current UVTools' erode fix for elephant foot is good, the double exposure trick doesn't work well for bottom layers, I apply it only for Normal layers. A bottom layer takes ExposureTime+extra time with one single erode layer.

**Reply by human890209** [reply]
Any dim pixel loses detail. The UV light blurs. I only apply dim pixels for infills, I never dim a wall pixel.

**Reply by sn4k3** [reply]
Depends on printer and light source.
At some point a dim percentage will match a full white at n exposure time, it's not linear but if linear 255 at 2s would be the same as 128 at 4s

**Reply by human890209** [reply]
Yes, I tested my QIDI 6.08 Mono screen.
I use Exposure Time Finder with Pixel Dimming to find the exposure time of several dim values, 255, 191, 159, 127. The exposure time is not linear and all dimed ones lose 1~2px details. I guess the results are caused by the light scattering of a dimed pixel.
